trx/tests/test_memmap.py

```diff
@@ -178,12 +174,117 @@ def test_load_directory(path):
     assert isinstance(tmm.load_from_directory(path), tmm.TrxFile)
 
 
-def test_concatenate():
-    pass
+@pytest.mark.parametrize("path", [("small.trx")])
+def test_concatenate(path):
+    path = os.path.join(get_home(), 'memmap_test_data', path)
+    trx1 = tmm.load(path)
+    trx2 = tmm.load(path)
+    concat = tmm.concatenate([trx1, trx2])
+    assert len(concat) == 2 * len(trx2)
 
 
-def test_save():
-    pass
+@pytest.mark.parametrize("path", [("small.trx")])
+def test_resize(path):
+    path = os.path.join(get_home(), 'memmap_test_data', path)
+    trx1 = tmm.load(path)
+    concat = tmm.TrxFile(nb_vertices=1000000, nb_streamlines=10000,
+                         init_as=trx1)
+
+    tmm.concatenate([concat, trx1], preallocation=True, delete_groups=True)
+    concat.resize()
+    assert len(concat) == len(trx1)
+
+
+@pytest.mark.parametrize(
+    "path, buffer",
+    [
+        ("small.trx", 10000),
+        ("small.trx", 0)
+    ]
+)
+def test_append(path, buffer):
+    path = os.path.join(get_home(), 'memmap_test_data', path)
+    trx1 = tmm.load(path)
+    concat = tmm.TrxFile(nb_vertices=1, nb_streamlines=1,
+                         init_as=trx1)
+
+    concat.append(trx1, extra_buffer=buffer)
+    if buffer > 0:
+        concat.resize()
+    assert len(concat) == len(trx1)
+
+
+@pytest.mark.parametrize("path, buffer", [("small.trx", 10000)])
+def test_append_StatefulTractogram(path, buffer):
+    path = os.path.join(get_home(), 'memmap_test_data', path)
+    trx = tmm.load(path)
+    obj = trx.to_sft()
+    concat = tmm.TrxFile(nb_vertices=1, nb_streamlines=1, init_as=trx)
+
+    concat.append(obj, extra_buffer=buffer)
+    if buffer > 0:
+        concat.resize()
+    assert len(concat) == len(obj)
+
+
+@pytest.mark.parametrize("path, buffer", [("small.trx", 10000)])
+def test_append_Tractogram(path, buffer):
+    path = os.path.join(get_home(), 'memmap_test_data', path)
+    trx = tmm.load(path)
+    obj = trx.to_tractogram()
+    concat = tmm.TrxFile(nb_vertices=1, nb_streamlines=1, init_as=trx)
+
+    concat.append(obj, extra_buffer=buffer)
+    if buffer > 0:
+        concat.resize()
+    assert len(concat) == len(obj)
+
+
+@pytest.mark.parametrize("path, size, buffer", [("small.trx", 50, 10000),
+                                                ("small.trx", 0, 10000),
+                                                ("small.trx", 25000, 10000),
+                                                ("small.trx", 50, 0),
+                                                ("small.trx", 0, 0),
+                                                ("small.trx", 25000, 10000)])
+def test_from_lazy_tractogram(path, size, buffer):
+    DATA = {}
```

Comment:
I am a bit confused: why are you setting up this dict, rather than assigning these as variables?